lib/pengine/native.c

```diff
@@ -834,6 +841,164 @@ pe__resource_xml(pcmk__output_t *out, va_list args)
+        offset += snprintf(buffer + offset, LINE_MAX - offset, "FAILED");
+    } else {
+        const char *rsc_state = native_displayable_state(rsc, options);
+
+        offset += snprintf(buffer + offset, LINE_MAX - offset, "%s", rsc_state);
+    }
+
+    if(node) {
+        offset += snprintf(buffer + offset, LINE_MAX - offset, " %s", node->details->uname);
+
+        if (node->details->online == FALSE && node->details->unclean) {
+            flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset,
+                                   "%sUNCLEAN", comma_if(flagOffset));
+        }
+    }
+
+    if (options & pe_print_pending) {
+        const char *pending_task = native_pending_task(rsc);
+
+        if (pending_task) {
+            flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset,
+                                   "%s%s", comma_if(flagOffset), pending_task);
+        }
+    }
+
+    if (target_role) {
+        enum rsc_role_e target_role_e = text2role(target_role);
+
+        /* Ignore target role Started, as it is the default anyways
+         * (and would also allow a Master to be Master).
+         * Show if target role limits our abilities. */
+        if (target_role_e == RSC_ROLE_STOPPED) {
+            flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset,
+                                   "%sdisabled", comma_if(flagOffset));
+            rsc->cluster->disabled_resources++;
+
+        } else if (is_set(uber_parent(rsc)->flags, pe_rsc_promotable)
+                   && target_role_e == RSC_ROLE_SLAVE) {
+            flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset,
+                                   "%starget-role:%s", comma_if(flagOffset), target_role);
+            rsc->cluster->disabled_resources++;
+        }
+    }
+
+    if (is_set(rsc->flags, pe_rsc_block)) {
+        flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset,
+                               "%sblocked", comma_if(flagOffset));
+        rsc->cluster->blocked_resources++;
+
+    } else if (is_not_set(rsc->flags, pe_rsc_managed)) {
+        flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset,
+                               "%sunmanaged", comma_if(flagOffset));
+    }
+
+    if(is_set(rsc->flags, pe_rsc_failure_ignored)) {
+        flagOffset += snprintf(flagBuffer + flagOffset, LINE_MAX - flagOffset,
+                               "%sfailure ignored", comma_if(flagOffset));
+    }
+
+    if ((options & pe_print_rsconly) || g_list_length(rsc->running_on) > 1) {
+        desc = crm_element_value(rsc->xml, XML_ATTR_DESC);
+    }
+
+    CRM_LOG_ASSERT(offset > 0);
+    if(flagOffset > 0) {
+        fprintf(out->dest, "%s (%s)%s%s", buffer, flagBuffer, desc?" ":"", desc?desc:"");
+    } else {
+        fprintf(out->dest, "%s%s%s", buffer, desc?" ":"", desc?desc:"");
+    }
+
+    if (g_list_length(rsc->running_on) > 1) {
```

Comment:
I think we still need to skip this if pe_print_rsconly is set

Reply:
Done.